woocommerce/woocommerce-gutenberg-products-block
# [Product Collection] Add Price Range Filter

## What

Adds a Price Range Filter to the Product Collection Block's Inspector Controls
<img src="https://github.com/woocommerce/woocommerce-blocks/assets/16707866/62e10296-ec53-49ad-a3ed-a47fbd628b5d" width="300">


- **Integration of Price Range Filter**: A `priceRange` attribute has been added to the block's attribute, allowing merchants to define a price range for displaying products. This is particularly useful for creating more targeted and specific product displays based on pricing criteria.
- **Back-end Support and Filtering Logic**: Corresponding back-end adjustments in `ProductCollection.php` ensure that the product query accounts for the new price range filter, effectively filtering products within the specified price boundaries.
- **Use of 'Auto' for Price Values**: The implementation allows for dynamic price range selection with an 'Auto' placeholder. When the price fields are left blank (auto), the filter dynamically adjusts the range to include all products, offering flexibility in how the price range is applied.

Fixes woocommerce/woocommerce#42377 

## Why


This update significantly improves the versatility of the Product Collection block, empowering merchants to curate products more effectively based on price ranges, and thus enhancing the shopping experience for customers.


## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Insert a product collection block into a new post.
2. From the Filters dropdown, select the “Price Range” filter. It will appear as shown:

<img src="https://github.com/woocommerce/woocommerce-blocks/assets/16707866/5a9e6a14-4aa1-4c00-abbf-6c3280524f2a" width="300">

3. Adjust the MIN & MAX values. Ensure that the products displayed in the Editor are filtered according to these updated values.
    - **Note:** The changes will only take effect if you *hit ENTER* or *click elsewhere*. This is designed to prevent unnecessary API calls while the merchant is adjusting the value.
4. Save the modifications and confirm that the same updates are visible on the Frontend.
5. Ensure that you can reset the filter's value by clicking in two different places, as indicated in the screenshot below:
<img src="https://github.com/woocommerce/woocommerce-blocks/assets/16707866/d6f2bf7f-0335-4f15-a28b-b618963a5e7a" width="300">


## Screenshots or screencast



| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility


Required:


## Checklist

Required:


Conditional:


## Changelog


> Product Collection - Add Price Range Filter
